package/vmalert.service

```diff
@@ -0,0 +1,16 @@
+[Unit]
+Description=Alerting for high-performance, cost-effective and scalable time series database, long-term remote storage for Prometheus.
+After=network.target
+
+[Service]
+Type=simple
+StartLimitBurst=5
+StartLimitInterval=0
+Restart=on-failure
+RestartSec=1
+ExecStart=/usr/bin/vmalert-prod -datasource.url=http://localhost:8428 -notifier.url=http://127.0.0.1:9093 -remoteRead.url=http://127.0.0.1:8428 -remoteWrite.url=http://127.0.0.1:8428 -rules=/etc/prometheus/rules.yaml
```

Comment:
ExecStart should not have args in the service file. usually you put those in a `/etc/default/myservice` file

in the service file you should have
```
[Service]
EnvironmentFile=/etc/default/mysservice
```

which would contain
```
###
# documentation about args
SERVICE_ARGS=<my parameters>
```

and the execstart directive could become
```
[Service]
ExecStart=/usr/bin/myservice $SERVICE_ARGS
```

Reply:
Thanks for the snippets, very nice.

Moving launch string to /etc/default would probably break RPM packaing and/or make it inconsistent. I did this PR as a work project and currently do not have time/space for proper refactoring. It was done in line with current package and because I found it useful (even though it is not ideal) I thought it might help someone. I however do not have time & space to properly refactor deb packaging (with debhelpers and such). Even more for RPM - but I would like not to break it.

Given that systemd units can be easily overridden by systemctl edit I would tend to leave it to this command.

Reply:
I suppose you're right about systemd overrides.

My only counterpoint  if you want to remain compatible with exotic or old distributions (i.e. with SystemV) `/etc/default/<service>` files are the norm. Staying on this old way of handling args is still, amha, quite a good practice. But i do agree that this won't stay like that for ever. And since systemd handles envfile quite well why not ? :)

Reply:
> My only counterpoint if you want to remain compatible with exotic or old distributions (i.e. with SystemV)

As I wrote in previous thread and as a user of exotic (is it really? :smile:) user of SystemV Debian I 100% agree. The reason was that RPM packages would either break or I would need to refactor them too (and as I said, currently no time to test) which seems worse to me that what I propose in PR. I was thinking of

```
EnvironmentFile=-/etc/default/victoria-metrics
EnvironmentFille=-/etc/sysconfig/victoria-metrics
```

Not sure if it is correct for RPM based distros and how to add conffiles to RPMs (for now). What do you think?

Reply:
For RPM distr work EnvironmentFile=-/etc/default/victoria-metrics